falcon/middleware.py

```diff
@@ -120,7 +120,11 @@ def process_response(
                 'Access-Control-Request-Headers', default='*'
             )
 
-            resp.set_header('Access-Control-Allow-Methods', allow)
+            if allow is not None:
+                # NOTE: not sure if it's more appropriate to raise an exception here
+                # This can happen only when a responder class defines a custom
+                # on_option responder method and does not set the 'Allow' header.
+                resp.set_header('Access-Control-Allow-Methods', allow)
```

Comment:
Ideas on what to do here? cc @vytas7 @kgriffs ?

Reply:
OK, so what are our `OPTIONS` (pun intended)?

First, I think middleware is for adding _simple_ CORS, so I think we should minimize the chances of it raising an exception.

Realistically, I think there two main alternatives:
* If `Allow` is empty or missing, assume that no methods are allowed, and disable CORS altogether for the response in question (delete all `Access-Control` headers that this middleware sets).
* If `Allow` is empty or missing, assume that all methods are allowed, and set it to `Access-Control-Allow-Methods`: `*`.

At least [`Allow` docs on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow) state:
> An empty Allow header indicates that the resource allows no request methods, which might occur temporarily for a given resource, for example.

Which dovetails into the first option, strip `Access-Control-*` headers from response if `Allow` is empty.

But for me both alternatives are acceptable, as long as we document the chosen behaviour clearly.

Reply:
Please also note that this may affect sinks as well. Not sure about generic sinks, but I think we do want to allow permissive CORS on static routes, if the user has requested it via that parameter or CORS middleware. I think it is a somewhat common usecase in development, to serve some frontend resources from a Falcon app via a static route.

Reply:
It doesn't seem that static route is currently adding allow headers. are you proposing to make it set the header?

Reply:
Yes, I think CORS should apply to static routes too.

Reply:
it may make sense to do this change in another route?

may make sense to revert the behaviour change here and fix it there too

Reply:
sorry I meant PR

Reply:
Yeah, we can create a separate issue for this as well not to pollute this PR

Reply:
will be handled in #2325